src/qml/QuickPanelPage.qml

```diff
@@ -0,0 +1,952 @@
+                        labelText: ""
+                    });
+                }
+            }
+        }
+        //% "Options"
+        slotModel.append({ type: "label", labelText: qsTrId("id-options"), toggleId: "", listView: "" });
+        //% "Battery Meter aligned to bottom?"
+        slotModel.append({ type: "config", labelText: qsTrId("id-battery-bottom"), toggleId: "", listView: "" });
+        //% "Enable colored battery?"
+        slotModel.append({ type: "config", labelText: qsTrId("id-battery-colored"), toggleId: "", listView: "" });
+        //% "Show battery charge animation?"
+        slotModel.append({ type: "config", labelText: qsTrId("id-battery-animation"), toggleId: "", listView: "" });
+        //% "Tap to cycle particle design"
+        slotModel.append({ type: "cycler", labelText: qsTrId("id-particle-design"), toggleId: "", listView: "" });
+        //% "Battery preview"
+        slotModel.append({ type: "display", labelText: qsTrId("id-battery-preview"), toggleId: "", listView: "" });
+
+        saveConfiguration();
+        listLoader.active = true;
+    }
+
+    // Count the number of fixed toggles in the model
+    function countFixedToggles() {
+        var count = 0;
+        for (var i = 1; i < slotModel.count; i++) {
+            if (slotModel.get(i).type === "toggle" && slotModel.get(i).listView === "fixed") {
+                count++;
+            }
+            if (slotModel.get(i).type === "label" && slotModel.get(i).labelText === qsTrId("id-sliding-row")) {
+                break;
+            }
+        }
+        return count;
+    }
+
+    // Check if a toggle is in the fixed row
+    function isToggleInFixedRow(toggleId) {
+        for (var i = 1; i < slotModel.count; i++) {
+            var item = slotModel.get(i);
+            if (item.type === "label" && item.labelText === qsTrId("id-sliding-row")) {
+                break;
+            }
+            if (item.type === "toggle" && item.toggleId === toggleId) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // Find the index of the slider label
+    function findSliderLabelIndex() {
+        for (var i = 0; i < slotModel.count; i++) {
+            if (slotModel.get(i).type === "label" && slotModel.get(i).labelText === qsTrId("id-sliding-row")) {
+                return i;
+            }
+        }
+        return 3;
+    }
+
+    // Find the index of the options label
+    function findOptionsLabelIndex() {
+        for (var i = 0; i < slotModel.count; i++) {
+            if (slotModel.get(i).type === "label" && slotModel.get(i).labelText === qsTrId("id-options")) {
+                return i;
+            }
+        }
+        return slotModel.count - 6;
+    }
+
+    // Ensure the slider label stays at the correct position
+    function ensureSliderLabelPosition() {
+        var sliderIndex = findSliderLabelIndex();
+        if (sliderIndex !== 3) {
+            slotModel.move(sliderIndex, 3, 1);
```

Comment:
Again why `3` here? Is it to ensure that the slider label is after position `2` (i.e. the fixed toggles)?